shared-functions.sh

```diff
@@ -80,16 +80,15 @@ ensure_mac_os() {
 
 # Mac-specific function to install Java JDK
 install_mac_java() {
-    # Determine which Java JDKs we have.
-    # If a non-Adopt Open JDK that is v1.8 (aka Java 8), that's ok!
-    java_versions=$(/usr/libexec/java_home --version "1.8" >/dev/null 2>&1 || echo "Not found")
-
-    if [ "$java_versions" = "Not found" ]; then
-        echo "Installing Adopt Open JDK v8..."
-        brew install homebrew/cask-versions/adoptopenjdk8
-    else
-        echo "java8 already installed ($java_versions)"
+    # It's suprisingly difficult to tell what java versions are
+    # already installed -- there are different java providers -- so we
+    # just always try to install the one we want.  For more info, see
+    #   https://github.com/Khan/khan-dotfiles/pull/61/files#r964917242
+    echo "Installing Adopt Open JDK v11..."
+    if ! brew tap | grep -qi "AdoptOpenJDK/openjdk"; then
+        brew tap AdoptOpenJDK/openjdk
     fi
+    brew install homebrew/cask-versions/adoptopenjdk11
```

Comment:
@MiguelCastillo do you mind trying this incantation out, both with a java11 installed already and without?

Reply:
sure thing!!  just a moment

Reply:
`brew install homebrew/cask-versions/adoptopenjdk11` isn't working (it can't find the cask)...  It should work with only `brew install adoptopenjdk11` tho.